pkg/awsutils/awsutils.go

```diff
@@ -248,10 +248,7 @@ func (ss *StringSet) Set(items []string) {
 }
 
 func (ss *StringSet) IsEmpty() bool {
-	if ss.data.Len() == 0 {
-		return true
-	}
-	return false
+	return ss.data != nil && ss.data.Len() == 0
```

Comment:
I think this method needs to hold the lock too (there are potential races between the two references to `ss.data`, and (possibly) internally in `Len()`)

Reply:
Updated to hold a Rlock here.